src/pages/index.js

```diff
@@ -49,10 +99,53 @@ constants.addNewCardButton.addEventListener("click", () => {
 });
 
 function handleImageSubmit(card) {
-  renderCard(card);
-  addNewCardModal.close();
-  constants.imageSubmitForm.reset();
-  addImageFormValidator.resetValidation();
+  addNewCardModal.setLoading(true, "Saving...");
+  api
+    .addCard(card)
+    .then(() => {
+      renderCard(card);
+    })
+    .finally(() => {
+      addNewCardModal.setLoading(false);
+      constants.imageSubmitForm.reset();
```

Comment:
Needs correction: Сlose modal and reset the form and the validation in `then` block, otherwise the form will close even if an error occurred and you will need to reopen the form and fill the values again. (fix that here and in all the other places)

Reply:
Needs correction: please move lines 121-123 to `then` block, which is located on line 116